### DISCUSSION TITLE
How to document JS Classes (not web components) using the JSDoc in an MDX

### DISCUSSION THREAD

**Discussion by Th3S4mur41** [storybookjs/storybook#17646] [Help] [answered]
Next to web components, our library also defines some JS Classes that we would also like to have documented in Storybook.
The current idea to automate this as much as possible would be to generate the docs based on JSDoc in an MDX page, but not sure how to achieve this?!

**Comment by Th3S4mur41** [comment]
We figured out a way...
We use the JSDoc to generate Markdown documentation with [typedoc](https://typedoc.org/).
Then we import the .md file into a story

```jsx
/** 
 * MyClass.stories.mdx 
 */

import { Description } from '@storybook/addon-docs';
import { Meta } from '@storybook/addon-docs/blocks';
import MyClass from '!raw-loader!docs/build/MyClass.md';

<Meta title="Storybook/Classes/MyClass" />

<Description markdown={MyClass} />
```

**Reply by braddialpad** [reply, marked as answer]
Update on this for Storybook 7 since the `<Description markdown>` syntax is no longer supported. The above example would now be like.

```mdx
import { Markdown, Meta } from '@storybook/blocks';
import MyClass from 'docs/build/MyClass.md?raw';

<Meta title="Storybook/Classes/MyClass" />

<Markdown>{MyClass}</Markdown>
```

**Reply by Th3S4mur41** [reply]
True, I forgot to update the post
